monitoring/uss_qualifier/scd/data_interfaces.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional, List, Dict
+from monitoring.monitorlib.typing import ImplicitDict
+from monitoring.monitorlib.scd_automated_testing.scd_injection_api import InjectFlightRequest
+from monitoring.uss_qualifier.common_data_definitions import Severity
+
+class Result(ImplicitDict):
+    ''' A class to hold the expected output from data processing and the test interface of the USS under test '''
+    processing_outcome: str
+        
+class KnownIssueFields(ImplicitDict):
+    ''' A class to hold a message that the test executor can provide to the USS in cases when the USS provides a response that is not same as the expected result. '''    
+    test_code: str
+    '''Code corresponding to check generating this issue'''
+
+    relevant_requirements: List[str] = []
+    '''Requirements that this issue relates to'''
+
+    severity: Severity
+    '''How severe the issue is'''
+
+    subject: Optional[str]
+    ''' Identifier of the subject of this issue, if applicable. This may be a UAS serial number, or any field of other object central to the issue. '''
+
+    summary: str
+    '''Human-readable summary of the issue'''
+
+    details: str
+    '''Human-readable description of the issue'''
+
+class KnownResponses(ImplicitDict):
+    ''' A class to evaluate results / response to an injection of test flight data (TestFlightRequest). This data structure holds information about the "result" response provided by the test interface of the USS under test and details of how long the executor should wait for a response. Per the SCD Testing API an acceptable result should be one of "Planned", "Rejected", "ConflictWithFlight" or "Failed". In the case where the USS under test provides a result that is not the same as expected result then the test driver can provide a message detailing why the expected result was populated the way it was. This should help the USS under test debug specific parts of their internal systems. '''
+    acceptable_results: List[str] 
+    incorrect_result_details: Optional[Dict[str, KnownIssueFields]]
+    response_wait_time_secs: int 
+    response_timeout_message: KnownIssueFields
+
+class InjectionTarget(ImplicitDict):
+    ''' A class to hold sequential assignment of the USS as per the SCD test API '''
```

Comment:
What part of the SCD test API does "as per the SCD test API" refer to?  It seems like the choice of identifying an injection target using an integer sequence number is being made here, not in the API.  `FlightInjectionAttempt.injection_target` needs to refer to a particular USS, but test executor will perform the mapping between "particular USS" and the URL/identity of that USS.  A sequence number is only one way to uniquely identify a particular USS in an `AutomatedTest`, and it is arguably not as good as using a more human-legible role name.  For instance, `InjectionTarget`'s sole field could be `uss_role: str` so that an automated test definition would look like

```json
"injection_attempts": [
  {
    "injection_target": {
      "uss_role": "Good USS"
    }
    "test_injection": {...},
    "known_responses": {...}
  },
  {
    "injection_target": {
      "uss_role": "Querying USS"
    }
    "test_injection": {...},
    "known_responses": {...}
  },
  {
    "injection_target": {
      "uss_role": "Good USS"
    }
    "test_injection": {...},
    "known_responses": {...}
  }
]
```

rather than

```json
"injection_attempts": [
  {
    "injection_target": {
      "uss_sequence_number": 1
    }
    "test_injection": {...},
    "known_responses": {...}
  },
  {
    "injection_target": {
      "uss_sequence_number": 2
    }
    "test_injection": {...},
    "known_responses": {...}
  },
  {
    "injection_target": {
      "uss_sequence_number": 1
    }
    "test_injection": {...},
    "known_responses": {...}
  }
]
```

Reply:
Thanks yes! It makes much more sense the way you have suggested. The sequence 1, 2 was referring to the [block diagram](https://github.com/interuss/dss/tree/master/interfaces/automated-testing/scd/design/astm-strategic-coordination) for nominal test. Your suggested way is much better.